lib/components/base/Slider.vue

```diff
@@ -0,0 +1,180 @@
+      class="slider-input"
+      @input="onTextInput"
+    />
+  </div>
+</template>
+
+<script>
+export default {
+  name: "Slider",
+  props: {
+    value: {
+      type: Number,
+      default: 0
+    },
+    min: {
+      type: Number,
+      default: 0
+    },
+    max: {
+      type: Number,
+      default: 100
+    },
+    step: {
+      type: Number,
+      default: 10
+    },
+    forceStep: {
+      type: Boolean,
+      default: false
+    }
+  },
+  emits: ['input'],
+  data() {
+    return {
+      sliderWidth: 0,
+      objectPosition: 0,
+      startOffset: 0,
+      currentValue: Math.max(this.min, this.value).toString()
+    };
+  },
+  computed: {
+    inputValueValid: {
+      get() {
+        return this.currentValue
+      },
+      set(newValue) {
+        if (newValue < this.min) {
+          this.currentValue = this.min.toString()
+        } else if (newValue > this.max) {
+          this.currentValue = this.max.toString()
+        } else if (!newValue) {
+          this.currentValue = this.min.toString()
+        } else {
+          this.currentValue = (newValue - (this.forceStep ? newValue % this.step : 0)).toString()
+        }
+      }
+    }
+  },
+  methods: {
+    onInput() {
+      // this.currentValue is a string because HTML is weird
+      this.$emit('input', parseInt(this.currentValue));
+    },
+    onTextInput() {
+      this.inputValueValid = parseInt(this.currentValue);
+    },
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.root-container {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+}
+
+.slide-container .slider {
+  width: 12rem;
+}
+
+.slider-component .slide-container {
+  width: 100%;
+}
+
+.slider-component .slide-container .slider {
+  -webkit-appearance: none;
+  appearance: none;
+  border-radius: 2px;
+  height: 4px;
+  background:  linear-gradient(
+      to right,
+      var(--color-brand),
+      var(--color-brand) calc((var(--current-value) - var(--min-value)) / (var(--max-value) - var(--min-value)) * 100%),
+      #c2c2c2 calc((var(--current-value) - var(--min-value)) / (var(--max-value) - var(--min-value)) * 100%),
+      #c2c2c2 100%
+  );
+  background-size: 100% 100%;
+  outline: none;
+  vertical-align: middle;
+}
+
+.slider-component .slide-container .slider::-webkit-slider-thumb {
+  -webkit-appearance: none;
+  appearance: none;
+  width: 12px;
+  height: 12px;
+  background: var(--color-brand);
+  cursor: pointer;
+  border-radius: 50%;
+  transition: .2s;
+}
+
+.slider-component .slide-container .slider::-moz-range-thumb {
+  width: 12px;
+  height: 12px;
+  background: var(--color-brand);
+  cursor: pointer;
+  border-radius: 50%;
+  transition: .2s;
+}
+
+.slider-component .slide-container .slider:hover::-webkit-slider-thumb {
+  width: 18px;
+  height: 18px;
+  transition: .2s;
+}
+
+.slider-component .slide-container .slider:hover::-moz-range-thumb {
+  width: 18px;
+  height: 18px;
+  transition: .2s;
+}
+
+.slider-input {
+  width: 100px;
```

Comment:
use rem units- also should it be limited? i feel like by default the width should be automatic and if the parent container is constrained, it should use that,